src/formatDate.js

```diff
@@ -8,7 +8,36 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dataParts = date.split(fromFormat[fromFormat.length - 1]);
```

Comment:
The logic assumes that the last element of `fromFormat` is the separator, which might not always be the case. Consider explicitly defining the separator or ensuring that the format arrays are structured to always have the separator as the last element.